encoder/csv.go

```diff
@@ -178,20 +141,27 @@ func NewCSVItemIterator(sourceConfig map[string]any, data io.ReadCloser) (*CSVIt
 	if ok {
 		reader.columns = columnNames.([]string)
 	}
-	// check if header is present and if so, validate amount fields
-	// do we want to do this here?
+	validateFields, ok := sourceConfig["validateFields"]
+	if ok {
+		if !validateFields.(bool) {
+			// if false
+			reader.decoder.FieldsPerRecord = -1
+		} else {
+			//checks the first field and sees how many columns there is, uses that as validation going forward.
+			//another option is to explicity set the number of columns in the config
+			reader.decoder.FieldsPerRecord = 0
```

Comment:
Thought we discussed this being the len of columns?

Reply:
We discussed it, but didn't we agree we didn't want a schema on incoming? The field is now set to empty string if nothing exists in it.

Reply:
disregard this, and see updated pr - didn't add latest changes to the pr